File: app/src/main/java/com/adityap/flashy_createflashcards/models/DeckModel.java
package com.adityap.flashy_createflashcards.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Aditya on 12/26/2018.
 */

    public class DeckModel implements Parcelable {
        private int mId;
        private String mDeckName;
        private String mDeckDescription;



        public DeckModel(String deckName, String deckDescription){
            mDeckName = deckName;
            mDeckDescription = deckDescription;
        }

    protected DeckModel(Parcel in) {
        mId = in.readInt();
        mDeckName = in.readString();
        mDeckDescription = in.readString();
    }

    public static final Creator<DeckModel> CREATOR = new Creator<DeckModel>() {
        @Override
        public DeckModel createFromParcel(Parcel in) {
            return new DeckModel(in);
        }

        @Override
        public DeckModel[] newArray(int size) {
            return new DeckModel[size];
        }
    };

    public int getId() {
            return mId;
        }

        public void setId(int mId) {
            this.mId = mId;
        }

        public String getDeckName() {
            return mDeckName;
        }

        public void setDeckName(String mDeckName) {
            this.mDeckName = mDeckName;
        }

        public String getDeckDescription() {
            return mDeckDescription;
        }

        public void setDeckDescription(String mDeckDescription) {
            this.mDeckDescription = mDeckDescription;
        }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(mId);
        dest.writeString(mDeckName);
        dest.writeString(mDeckDescription);
    }
}


//    public String getDeckName() {
//        return mDeckName;
//    }
//
//    public void setDeckname(String mDeckName) {
//        this.mDeckName = mDeckname;
//    }
//
//    public String getDeckDescription() {
//        return mDeckDescription;
//    }
//
//    public void setDeckDescription(String mDeckDescription) {
//        this.mDeckDescription = mdeckDescription;
//    }

File: app/src/main/java/com/adityap/flashy_createflashcards/models/DeckModel.kt
package com.adityap.flashy_createflashcards.models

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by Aditya on 12/26/2018.
 */
data class DeckModel(val id : Int = 0, val deckName: String, val deckDescription: String?) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            requireNotNull(parcel.readString()),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
        parcel.writeString(deckName)
        parcel.writeString(deckDescription)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<DeckModel> {
        override fun createFromParcel(parcel: Parcel): DeckModel {
            return DeckModel(parcel)
        }

        override fun newArray(size: Int): Array<DeckModel?> {
            return arrayOfNulls(size)
        }
    }
}
